### PR TITLE
Home page

### PR DESCRIPTION
**Pull request by Razor0744**
Changed libs for navigation, added repository, usecase and HomeScreen with HomeViewModel

### CODE REVIEW

File: app/src/main/java/com/example/movopfy/di/AppDi.kt [hunk 1]
```diff
@@ -1,5 +1,11 @@
 package com.example.movopfy.di
 
+import com.example.movopfy.features.home.presentation.HomeViewModel
+import org.koin.androidx.viewmodel.dsl.viewModel
 import org.koin.dsl.module
 
-val appModule = module { }
\ No newline at end of file
+val appModule = module {
```
Comment (MaximSemashko) [inline app/src/main/java/com/example/movopfy/di/AppDi.kt]: In your home package create koin package, add val homeModule = module {...} and place all classes registration in that place that related to home (service, repository, viewmodel). Please remove dataDi, domainDi.

File: app/src/main/java/com/example/movopfy/features/home/data/api/anilibria/Anilibria.kt [hunk 2]
```diff
@@ -1,26 +1,29 @@
...
-    suspend fun getSchedule(): List<AnilibriaSchedule> = suspendCoroutine {
-        anilibriaURL.anilibriaAPI.getSchedule().enqueue(object :
-            retrofit2.Callback<List<AnilibriaSchedule>> {
-            override fun onResponse(
-                call: Call<List<AnilibriaSchedule>>,
-                response: Response<List<AnilibriaSchedule>>
-            ) {
-                it.resume(response.body() as List<AnilibriaSchedule>)
-            }
+//    suspend fun getSchedule(): List<AnilibriaSchedule> = suspendCoroutine {
```
Comment (MaximSemashko) [inline app/src/main/java/com/example/movopfy/features/home/data/api/anilibria/Anilibria.kt]: Please remove unused code

File: app/src/main/java/com/example/movopfy/features/home/data/api/anilibria/AnilibriaService.kt [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+import com.example.movopfy.features.home.data.models.AnilibriaSchedule
+import retrofit2.Response
+import retrofit2.http.GET
+
+interface AnilibriaService {
+
+    @GET("title/schedule")
+    suspend fun getSchedule(): Response<List<AnilibriaSchedule>>
+
```
Comment (MaximSemashko) [inline app/src/main/java/com/example/movopfy/features/home/data/api/anilibria/AnilibriaService.kt]: remove empty line

File: app/src/main/java/com/example/movopfy/features/home/data/api/anilibria/AnilibriaURL.kt [hunk 4]
```diff
@@ -5,6 +5,6 @@ import com.example.movopfy.retrofit.RetrofitClient
 class AnilibriaURL(private val retrofitClient: RetrofitClient) {
```
Comment (MaximSemashko) [inline app/src/main/java/com/example/movopfy/features/home/data/api/anilibria/AnilibriaURL.kt]: That class should be removed. Will be disscused on a call

File: app/src/main/java/com/example/movopfy/features/home/data/repository/AnilibriaRepositoryImpl.kt [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+
+import com.example.movopfy.features.home.data.api.anilibria.Anilibria
+import com.example.movopfy.features.home.data.models.AnilibriaSchedule
+import com.example.movopfy.features.home.domain.repository.AnilibriaRepository
+
+class AnilibriaRepositoryImpl(private val anilibria: Anilibria) : AnilibriaRepository{
+    override suspend fun getSchedule(): List<AnilibriaSchedule> {
+        return anilibria.getSchedule()
+    }
+
```
Comment (MaximSemashko) [inline app/src/main/java/com/example/movopfy/features/home/data/repository/AnilibriaRepositoryImpl.kt:11]: please remove empty lines all over the project

File: app/src/main/java/com/example/movopfy/features/home/data/repository/AnilibriaRepositoryImpl.kt [hunk 6]
```diff
@@ -0,0 +1,12 @@
+package com.example.movopfy.features.home.data.repository
+
+import com.example.movopfy.features.home.data.api.anilibria.Anilibria
+import com.example.movopfy.features.home.data.models.AnilibriaSchedule
+import com.example.movopfy.features.home.domain.repository.AnilibriaRepository
+
+class AnilibriaRepositoryImpl(private val anilibria: Anilibria) : AnilibriaRepository{
+    override suspend fun getSchedule(): List<AnilibriaSchedule> {
+        return anilibria.getSchedule()
```
Comment (MaximSemashko) [inline app/src/main/java/com/example/movopfy/features/home/data/repository/AnilibriaRepositoryImpl.kt]: surrond that code with withContext(Dispatchers.IO)

File: app/src/main/java/com/example/movopfy/features/home/domain/usecase/GetWaitingListTodayUseCase.kt [hunk 7]
```diff
@@ -0,0 +1,23 @@
+package com.example.movopfy.features.home.domain.usecase
+
+import com.example.movopfy.features.home.domain.models.WaitingListToday
+import com.example.movopfy.features.home.domain.repository.AnilibriaRepository
+
+class GetWaitingListTodayUseCase(private val anilibriaRepository: AnilibriaRepository) {
+
+    suspend fun execute(day: Int): List<WaitingListToday> {
+        val schedule = anilibriaRepository.getSchedule()
+        val waitingListToday = arrayListOf<WaitingListToday>()
```
Comment (MaximSemashko) [inline app/src/main/java/com/example/movopfy/features/home/domain/usecase/GetWaitingListTodayUseCase.kt]: can be used just mutableListOf

File: app/src/main/java/com/example/movopfy/features/home/domain/usecase/GetWaitingListTodayUseCase.kt [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+
+        for (i in schedule[day].list) {
```
Comment (MaximSemashko) [inline app/src/main/java/com/example/movopfy/features/home/domain/usecase/GetWaitingListTodayUseCase.kt]: please convert to map

File: app/src/main/java/com/example/movopfy/features/home/presentation/HomeScreen.kt [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+import com.example.movopfy.features.home.domain.models.WaitingListToday
+import org.koin.androidx.compose.koinViewModel
+
+@Composable
+fun HomeScreen(viewModel: HomeViewModel = koinViewModel()) {
+
+    val uiState by viewModel.uiState.collectAsState()
+
+    when (uiState) {
+        is HomeViewModel.HomeUiState.Loading -> {}
```
Comment (MaximSemashko) [inline app/src/main/java/com/example/movopfy/features/home/presentation/HomeScreen.kt]: Please add ProgressBar

File: app/src/main/java/com/example/movopfy/features/home/presentation/HomeScreen.kt [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+        is HomeViewModel.HomeUiState.Loaded -> {
+            val list = (uiState as HomeViewModel.HomeUiState.Loaded).schedules
```
Comment (MaximSemashko) [inline app/src/main/java/com/example/movopfy/features/home/presentation/HomeScreen.kt]: You dont need to do like that.

```
when (val state = uiState) {
...
-> {
val list = state.schedules
}
}
```

File: app/src/main/java/com/example/movopfy/features/title/presentation/TitleScreen.kt [hunk 11]
```diff
@@ -0,0 +1,8 @@
+package com.example.movopfy.features.title.presentation
+
+import androidx.compose.runtime.Composable
+
+@Composable
+fun TitleScreen(){
```
Comment (MaximSemashko) [inline app/src/main/java/com/example/movopfy/features/title/presentation/TitleScreen.kt]: Unused code

File: app/src/main/java/com/example/movopfy/features/home/presentation/HomeScreen.kt [hunk 12]
```diff
@@ -0,0 +1,105 @@
...
+            }
+        }
+    }
+}
+
+@Composable
+fun TextWaitingListToday(dimensions: Dimensions) {
+
+    Text(
+        text = "Расписание на сегодня",
```
Comment (MaximSemashko) [inline app/src/main/java/com/example/movopfy/features/home/presentation/HomeScreen.kt]: please extract to string resources

File: app/src/main/java/com/example/movopfy/features/home/presentation/HomeScreen.kt [hunk 13]
```diff
@@ -0,0 +1,105 @@
...
+                modifier = Modifier
+                    .clip(shape = RoundedCornerShape(size = 15.dp))
+                    .clickable {  }
+            )
+        }
+    }
+}
+
+@Composable
+fun ProgressBarForWaitingListToday() {
```
Comment (MaximSemashko) [inline app/src/main/java/com/example/movopfy/features/home/presentation/HomeScreen.kt]: Please, move it uiComponents and rename to LoadingIndicator or LoadingState

File: app/src/main/java/com/example/movopfy/features/title/presentation/TitleScreen.kt (hunk 11 above)
Comment (Razor0744) [reply app/src/main/java/com/example/movopfy/features/title/presentation/TitleScreen.kt]: fixed

File: app/src/main/java/com/example/movopfy/features/home/presentation/HomeScreen.kt (hunk 12 above)
Comment (Razor0744) [reply app/src/main/java/com/example/movopfy/features/home/presentation/HomeScreen.kt]: I did name text_waiting_today_home_screen

File: app/src/main/java/com/example/movopfy/features/home/presentation/HomeScreen.kt (hunk 13 above)
Comment (Razor0744) [reply app/src/main/java/com/example/movopfy/features/home/presentation/HomeScreen.kt]: I did

File: app/src/main/java/com/example/movopfy/features/home/presentation/HomeScreen.kt (hunk 12 above)
Comment (MaximSemashko) [reply app/src/main/java/com/example/movopfy/features/home/presentation/HomeScreen.kt]: please rename to schedule_for_today_text

File: app/src/main/java/com/example/movopfy/di/AppDi.kt (hunk 1 above)
Comment (Razor0744) [reply app/src/main/java/com/example/movopfy/di/AppDi.kt]: did

File: app/src/main/java/com/example/movopfy/features/home/data/api/anilibria/Anilibria.kt (hunk 2 above)
Comment (Razor0744) [reply app/src/main/java/com/example/movopfy/features/home/data/api/anilibria/Anilibria.kt]: I did

File: app/src/main/java/com/example/movopfy/features/home/data/api/anilibria/AnilibriaService.kt (hunk 3 above)
Comment (Razor0744) [reply app/src/main/java/com/example/movopfy/features/home/data/api/anilibria/AnilibriaService.kt]: I did

File: app/src/main/java/com/example/movopfy/features/home/data/api/anilibria/AnilibriaURL.kt (hunk 4 above)
Comment (Razor0744) [reply app/src/main/java/com/example/movopfy/features/home/data/api/anilibria/AnilibriaURL.kt]: I did

File: app/src/main/java/com/example/movopfy/features/home/data/repository/AnilibriaRepositoryImpl.kt (hunk 5 above)
Comment (Razor0744) [reply app/src/main/java/com/example/movopfy/features/home/data/repository/AnilibriaRepositoryImpl.kt:11]: I did

File: app/src/main/java/com/example/movopfy/features/home/data/repository/AnilibriaRepositoryImpl.kt (hunk 6 above)
Comment (Razor0744) [reply app/src/main/java/com/example/movopfy/features/home/data/repository/AnilibriaRepositoryImpl.kt]: I did

File: app/src/main/java/com/example/movopfy/features/home/domain/usecase/GetWaitingListTodayUseCase.kt (hunk 7 above)
Comment (Razor0744) [reply app/src/main/java/com/example/movopfy/features/home/domain/usecase/GetWaitingListTodayUseCase.kt]: I think that use model for that better

File: app/src/main/java/com/example/movopfy/features/home/domain/usecase/GetWaitingListTodayUseCase.kt (hunk 8 above)
Comment (Razor0744) [reply app/src/main/java/com/example/movopfy/features/home/domain/usecase/GetWaitingListTodayUseCase.kt]: yes, that is better to look

File: app/src/main/java/com/example/movopfy/features/home/presentation/HomeScreen.kt (hunk 9 above)
Comment (Razor0744) [reply app/src/main/java/com/example/movopfy/features/home/presentation/HomeScreen.kt]: progress bar now in uiComponents

File: app/src/main/java/com/example/movopfy/features/home/presentation/HomeScreen.kt (hunk 10 above)
Comment (Razor0744) [reply app/src/main/java/com/example/movopfy/features/home/presentation/HomeScreen.kt]: I did

**Review by MaximSemashko — APPROVED** [review]

**Pull request merged**